### PR TITLE
#36 type validation

### PR DESCRIPTION
**Pull request by lezhnev74**
This PR contains updated type validation #36

### CODE REVIEW

File: src/Schema/Keywords/Type.php [hunk 1]
```diff
@@ -48,8 +49,7 @@ public function validate($data, string $type, ?string $format = null) : void
                     throw TypeMismatch::becauseTypeDoesNotMatch(CebeType::OBJECT, $data);
                 }
                 break;
-            case 'array':
-                // no constant here yet https://github.com/cebe/php-openapi/pull/24
+            case CebeType::ARRAY:
```
Comment (cebe) [inline src/Schema/Keywords/Type.php:48]: composer.json should require at least `"cebe/php-openapi": "^1.0.1"` as constant did not exist before that version.

Comment (cebe) [reply src/Schema/Keywords/Type.php:48]: there should be no reason to use anything below 1.2 though, so if you bump the constraint you can go with `"^1.2"` directly.

File: src/Schema/Keywords/Type.php [hunk 2]
```diff
@@ -39,7 +40,7 @@ public function validate($data, string $type, ?string $format = null) : void
     {
         switch ($type) {
             case CebeType::BOOLEAN:
-                if (! is_bool($data)) {
+                if (! is_bool($data) && ! preg_match('#^(true|false)$#i', (string) $data)) {
```
Comment (cebe) [inline src/Schema/Keywords/Type.php]: string cast on unknown type of $data may cause a notice: "Notice: Array to string conversion"

Comment (lezhnev74) [reply src/Schema/Keywords/Type.php]: Wow, I wouldn't have thought about that!

**Pull request merged**